### PR TITLE
Added an option to always use a defined database for KeePassNatMsg connection

### PR DESCRIPTION
**Pull request by berrnd**
Hi (again :)),

I've had the following problem:
I use KeePass always with multiple databases, but only the "first" is connected via KeePassNatMsg - it works as intended that also entries from the other open databases are returned, but only if the currently selected database is the connected one (as stated in the options dialog).

But that always gives the problem that when I did something in one of the other open databases and just minimized KeePass, so that the currently selected database is not the connected one, the connection does not work anymore.

So I added a config option to always use a defined database (by name) for the KeePassNatMsg connection, regardless of the currently selected one. When the config option is not set or contains an invalid / not currently opened database name, it uses the currently selected database - as of now.

This is how the changed options dialog looks like:

grafik

### CODE REVIEW

**Comment by smorks** [comment]
this is a good idea, thank you! i'll review this soon.

**Comment by smorks** [comment]
sorry for taking so long to get back to you on this! i've finally reviewed & tested this, and it all looks good.

i just have one suggestion:

instead of using the database name (which isn't required to fill out, and could have duplicates), maybe we store something like the database hash from the GetDbHash() function, or maybe just the RootGroup.Uuid in the option?

then, in the option dropdown, we could either put the DB Name if it's there, or maybe the filename or something along those lines, just so we don't end up with a dropdown with empty items?

what do you think?

**Comment by berrnd** [comment]
> sorry for taking so long to get back to you on this! i've finally reviewed & tested this, and it all looks good.

Absolutely no problem, thanks for looking into it. :)

Sounds all good, also thought about it - but didn't look deep enough into the KeePass plugin / data model and found no ID to use at first sight...

**Comment by berrnd** [comment]
Sorry for not coming back so long on this...

> instead of using the database name (which isn't required to fill out, and could have duplicates), maybe we store something like the database hash from the GetDbHash() function, or maybe just the RootGroup.Uuid in the option?

> then, in the option dropdown, we could either put the DB Name if it's there, or maybe the filename or something along those lines, just so we don't end up with a dropdown with empty items?

I did that now (and also rebased on your current master branch): The `GetDbHash()` of the selected connection DB is stored and if the DB name is empty the file path of the DB is shown in the dropdown.

**Comment by smorks** [comment]
awesome! thank you! i'll review this soon!

File: KeePassNatMsg/OptionsForm.cs [hunk 1]
```diff
@@ -64,7 +73,8 @@ private void okButton_Click(object sender, EventArgs e)
             _config.ReturnStringFieldsWithKphOnly = returnStringFieldsWithKphOnlyCheckBox.Checked;
             _config.SortResultByUsername = SortByUsernameRadioButton.Checked;
             _config.OverrideKeePassXcVersion = txtKPXCVerOverride.Text;
-            if (_restartRequired)
+			_config.ConnectionDatabaseHash = (comboBoxDatabases.SelectedItem as dynamic).DatabaseHash;
```
Comment (smorks) [inline KeePassNatMsg/OptionsForm.cs]: i get an error when blanking out the combo box. this appears to fix it:

```suggestion
			_config.ConnectionDatabaseHash = (comboBoxDatabases.SelectedItem as dynamic)?.DatabaseHash;
```

Comment (berrnd) [reply KeePassNatMsg/OptionsForm.cs]: Thanks - my bad, haven't tested it apparently with not selecting a connection database...

**Comment by smorks** [comment]
i will try and get this released in the next few days. thanks again!

**Pull request merged**